File: apps/python/lib/domain/type/fizz_buzz_type_03.py
from lib.domain.model.fizz_buzz_value import FizzBuzzValue
from lib.domain.type.fizz_buzz_type import FizzBuzzType


class FizzBuzzType03(FizzBuzzType):
    """タイプ 3: FizzBuzz のみ、それ以外は数を返す。"""

    def generate(self, number: int) -> FizzBuzzValue:
        if self._is_fizz_buzz(number):
            return FizzBuzzValue(number, "FizzBuzz")
        return FizzBuzzValue(number, str(number))

File: apps/php/src/Domain/Type/FizzBuzzType03.php
<?php

namespace App\Domain\Type;

use App\Domain\Model\FizzBuzzValue;

final class FizzBuzzType03 implements FizzBuzzType
{
    public function generate(int $number): FizzBuzzValue
    {
        if ($number % 15 === 0) {
            return new FizzBuzzValue($number, 'FizzBuzz');
        }

        return new FizzBuzzValue($number, (string) $number);
    }
}
